code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -138,3 +139,95 @@
 
 /obj/item/gun/energy/laser/redtag/hitscan
 	ammo_type = list(/obj/item/ammo_casing/energy/laser/redtag/hitscan)
+
+
+//////////////EXTERNAL LENS////////////////////////////
+
+/obj/item/external_lens
+	name = "external lens"
+	icon_state = "external"
+	desc = "external lens."
+	icon = 'icons/obj/guns/energy.dmi'
+	var/stored_ammo_type = /obj/item/ammo_casing/energy/laser
+	var/overlay = "laser"
+
+/obj/item/external_lens/Initialize()
+	. = ..()
+	add_overlay(overlay)
+
+/obj/item/external_lens/afterattack(atom/movable/AM, mob/user, flag)
+	. = ..()
+	if(istype(AM, /obj/item/gun/energy/laser))
+		var/obj/item/gun/energy/laser/L = AM
+		if(!L.modifystate)

```

Comment:
Please do not reuse this variable for this purpose. Use the component tools to handle what happens when multiple of the component is added to a gun

Reply:
i was using getcomponent to check if it had the laser component already, which works but this happens on something i havent touched http://take.ms/iO1AD

Reply:
That error is unrelated to anything you're doing here